src/wallet/wallet.cpp

```diff
@@ -4521,7 +4545,7 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(std::shared_ptr<CWallet>
         // Do the migration of keys and scripts for non-blank wallets, and cleanup if it fails
         success = local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET);
         if (!success) {
-            success = DoMigration(*local_wallet, context, error, res);
+            success = DoMigration(*local_wallet, context, error, res, empty_local_wallet);
         } else {
             // Make sure that descriptors flag is actually set
             local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
```

Comment:
Please feel free to disregard:

```suggestion
        if (!local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET)) {
            success = DoMigration(*local_wallet, context, error, res, empty_local_wallet);
        } else {
            success = true;
            // Make sure that descriptors flag is actually set
            local_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
```

Reply:
This has been resolved recently in #32149:

https://github.com/bitcoin/bitcoin/blob/a4fd565191e3184a174757f7a8fbda6afc98e50a/src/wallet/wallet.cpp#L4537-L4544